actionview/lib/action_view/lookup_context.rb

```diff
@@ -132,6 +132,11 @@ def exists?(name, prefixes = [], partial = false, keys = [], **options)
       end
       alias :template_exists? :exists?
 
+      def template_exists_in_any_format?(name, prefixes = [], partial = false, keys = [], **options)
+        options_with_all_formats = options.merge(formats: default_formats)
+        @view_paths.exists?(*args_for_lookup(name, prefixes, partial, keys, options_with_all_formats))
+      end
```

Comment:
:+1: to add more comments and maybe write the rules (https://github.com/rails/rails/issues/20666#issuecomment-116882771) somewhere explicitly in the code.

Regarding variants, I think the responsibility of this method should be to check if any template exists for a given `name` and `variant`. For each possible pair of ("template name", "variant"), I can easily apply @chancancode's rules to determine whether is the HTTP code, 406 or 204, for the response.

Does not it make sense for you, guys?